data/storeDEPRECATED/mongo/mongo.go

```diff
@@ -431,6 +433,13 @@ func (d *DataSession) CreateDataSetData(ctx context.Context, dataSet *upload.Upl
 		}
 		elapsed_time := time.Since(start).Milliseconds()
 		dataWriteToReadStoreMetrics.WithLabelValues("cbg").Observe(float64(elapsed_time))
+		// update meta data
+		if incomingUserMetadata != nil {
+			err = d.BucketStore.UpsertMetaData(ctx, dataSet.UserID, incomingUserMetadata)
```

Comment:
Not sure how to implement that but we may do the upsert only if incomingUserMetadata is modified (in order to avoid unnecessary writes).

Reply:
we can do it in the upsert Metadata method, as there is a function that "compare" the inccomingUserMetadate and the one in the db